### PR TITLE
chore: Added custom AnalyticsProvider test

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

Adds a basic unit test to exercise custom `AnalyticsProvider` functionality.

#### Description of how you validated changes
Local run.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/analytics/__tests__/Analytics.test.ts [hunk 1]
```diff
@@ -21,6 +22,31 @@ const record_spyon = jest
...
+class TestCustomProvider implements AnalyticsProvider {
+	getProviderName(): string {
+		return 'CustomProvider' as const;
+	}
+
+	getCategory() {
+		return 'Analytics' as const;
+	}
+
+	configure(o: any) {
```
Comment (kvramyasri7) [inline packages/analytics/__tests__/Analytics.test.ts:39]: just a question: is it okay to have params type as `any` when we had type as `Object`  in the interface ?

**Review by kvramyasri7 — DISMISSED** [review]
LGTM! Thanks @jimblanc

File: packages/analytics/__tests__/Analytics.test.ts [hunk 2]
```diff
@@ -21,6 +22,31 @@ const record_spyon = jest
 		return handlers.resolve();
 	});
 
+/**
+ * CAUTION: This mock class implements a publically available interface `AnalyticsProvider` which customers can use to
```
Comment (cshfang) [inline packages/analytics/__tests__/Analytics.test.ts:26]: Nit:
```suggestion
 * CAUTION: This mock class implements a publicly available interface `AnalyticsProvider` which customers can use to
```

File: packages/analytics/__tests__/Analytics.test.ts [hunk 3]
```diff
@@ -126,6 +152,26 @@ describe('Analytics test', () => {
...
+
+			analytics.addPluggable(customProvider);
+
+			const recordResponse = await analytics.record(
+				{ name: 'testEvent' },
+				'CustomProvider'
+			);
+
+			expect(customProviderRecordSpy).toBeCalled();
+			expect(recordResponse).toBeTruthy();
```
Comment (cshfang) [inline packages/analytics/__tests__/Analytics.test.ts:173]: Nit: Avoid using `toBeTruthy` when testing for boolean response
```suggestion
			expect(recordResponse).toBe(true);
```

**Review by cshfang — APPROVED** [review]

**Review by israx — APPROVED** [review]

**Review by kvramyasri7 — APPROVED** [review]

**Review by cshfang — APPROVED** [review]

**Pull request merged**